wallet-sdk/src/main/kotlin/org/stellar/walletsdk/customer/Customer.kt

```diff
@@ -0,0 +1,145 @@
+package org.stellar.walletsdk.customer
+
+import io.ktor.client.*
+import io.ktor.http.*
+import org.stellar.walletsdk.auth.AuthToken
+import org.stellar.walletsdk.exception.CustomerCreationException
+import org.stellar.walletsdk.exception.CustomerNotFoundException
+import org.stellar.walletsdk.exception.ErrorOnDeletingCustomerException
+import org.stellar.walletsdk.exception.UnauthorizedCustomerDeletionException
+import org.stellar.walletsdk.json.toJson
+import org.stellar.walletsdk.util.Util.authDelete
+import org.stellar.walletsdk.util.Util.authGet
+import org.stellar.walletsdk.util.Util.putJson
+
+class Customer
+internal constructor(
+  private val token: AuthToken,
+  private val baseUrl: String,
+  private val httpClient: HttpClient,
+) {
+  /**
+   * Get customer information by customer ID.
+   *
+   * @param id customer id
+   * @param type customer type
+   * @return a customer data object
+   */
+  suspend fun getByIdAndType(id: String, type: String): GetCustomerResponse {
+    val urlBuilder = URLBuilder(baseUrl)
+    urlBuilder.appendPathSegments("customer")
+    urlBuilder.parameters.append("id", id)
+    urlBuilder.parameters.append("type", type)
+    val urlString = urlBuilder.buildString()
+
+    val response = httpClient.authGet<GetCustomerResponse>(urlString, token)
+
+    if (response.id == null) {
+      throw CustomerNotFoundException(account = token.principalAccount)
+    }
+
+    return response
+  }
+
+  /**
+   * Create a new customer.
+   *
+   * @param sep9Info map of customer SEP-9 information. To create a new customer fields first_name,
+   * last_name, and email_address are required
+   * @param type (optional) the type of action the customer is being KYC for. See the Type
+   * Specification on SEP-12 definition.
+   * @return a customer with id information
+   */
+  suspend fun add(
+    sep9Info: Map<String, String>,
+    type: String? = null,
+  ): AddCustomerResponse {
+
+    var customer: Map<String, String> = mapOf()
+
+    if (type != null) {
+      customer = customer + mapOf("type" to type)
+    }
+
+    val isMissingFields =
+      sep9Info.isEmpty() ||
+        !sep9Info.containsKey("first_name") ||
```

Comment:
I don't think you need to check for any particular fields. Anchor decide what fields are optional and what are mandatory

Reply:
Anchor requires those fields. Otherwise, Anchor returns an error.

Reply:
Yes, testanchor requires it. Some arbitrary anchor won't require it.

Reply:
I removed that verification